src/platform/i386/boot_comp.c

```diff
@@ -159,12 +174,14 @@ kern_boot_comp(void)
 	 * Need to account for the pages that will be allocated as
 	 * PTEs
 	 */
-	untyped_pgd = (pgtbl_t)chal_va2pa(mem_boot_alloc(1));
-	if (pgtbl_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_UNTYPED_PT, untyped_pgd, 0)) assert(0);
+	if (pgtbl_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_UNTYPED_PT, pgtbl, 0)) assert(0);
 	nkmemptes = boot_nptes(mem_utmem_end() - mem_boot_end());
 	ret = boot_pgtbl_mappings_add(ct, BOOT_CAPTBL_SELF_UNTYPED_PT, BOOT_CAPTBL_KM_PTE, "untyped memory", mem_boot_nalloc_end(nkmemptes),
```

Comment:
As each page table reserves a range for the kernel, we add untyped memory starting from the address `BOOT_MEM_KM_BASE` (which is the first page currently). Is it possible to overwrite the kernel range in the page table?